asdf/asdftypes.py

```diff
@@ -25,6 +26,9 @@
                           list(six.integer_types) +
                           [float, list, dict, tuple])
 
+# regex used to parse module name from optional version string
+MODULE_RE = re.compile(r'([a-zA-Z]+)(-(\d.\d.\d))?')
```

Comment:
Not being a re expert, but isn't this pattern a little too narrow? What if version numbers get up to 10 or more?